pkg/virt-controller/watch/pool/pool_test.go

```diff
@@ -273,23 +274,17 @@ var _ = Describe("Pool", func() {
 			pool.Generation = 123
 			newPoolRevision := createPoolRevision(pool)
 
-			vm.Name = fmt.Sprintf("%s-0", pool.Name)
-			vm = injectPoolRevisionLabelsIntoVM(vm, poolRevision.Name)
-			vmi := createReadyVMI(vm, poolRevision)
-			markVmAsReady(vm)
-			watchtesting.MarkAsReady(vmi)
-			addPool(pool)
-			addVM(vm)
-			addVMI(vmi)
+			createVMsWithOrdinal(pool, 1, poolRevision, poolRevision, vm)
 			addCR(poolRevision)
 			// not expecting vmi to cause enqueue of pool because VMI and VM use the same pool revision
-			controller.vmiStore.Add(vmi)
-
 			expectControllerRevisionCreation(newPoolRevision)
-			expectVMUpdate(newPoolRevision.Name)
 
 			sanityExecute()
 
+			vm, err := fakeVirtClient.KubevirtV1().VirtualMachines(pool.Namespace).Get(context.TODO(), fmt.Sprintf("%s-0", pool.Name), metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vm.Labels).To(HaveKeyWithValue(v1.VirtualMachinePoolRevisionName, poolRevision.Name))
+
```

Comment:
Assert count of calls to `"update, "virtualmachineinstance"`?

Reply:
Same for other test cases

Reply:
Or remove such Expects alltogether if the exact outcome has been asserted, which is more realistic anyway.